apis/v1alpha1/route_types.go

```diff
@@ -35,6 +35,23 @@ const (
 	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
 )
 
+const (
+	// AnnotationAppProtocol defines the protocol a Gateway should use for
+	// communication with a Kubernetes Service. The annotation serves as a stop-gap
+	// solution for the Kubernetes versions < 1.18, where the `AppProtocol` field
+	// on the `Service` resource is not available. Using this annotation is highly
+	// discouraged when AppProtocol field is available. AppProtocol field, when
+	// populated, takes precedence over this annotation.
+	//
+	// The annotation value must be a comma separated with each value containing
+	// a port number followed by a forward slash and the protocol.
+	// Examples:
+	//
+	// - `networking.x-k8s.io/app-protocol: 8000/http,8443/https`
```

Comment:
We've defined our protocols as all uppercase. I know that's not ideal but it does match what's done in upstream. The suggestions for app protocol are noticeably lowercase here, contrasting with what we have elsewhere for HTTP. I'm not necessarily suggesting a change here, just calling out a difference.

Reply:
As @jpeach pointed out below, this lowercase approach matches the format of the AppProtocol field introduced in Kubernetes.

Reply:
For reference, we're using the IsQualifiedName function to validate AppProtocol in upstream (we are not validating against the full service names list but we are requiring that values be lowercase).

Reply:
Would it help if we document that the protocol is case-insensitive in here and examples are updated to be uppercase?